pkg/runtime/helper.go

```diff
@@ -17,12 +17,54 @@ limitations under the License.
 package runtime
 
 import (
+	"encoding/json"
 	"fmt"
 	"reflect"
+	"strings"
 
 	"k8s.io/kubernetes/pkg/conversion"
 )
 
+// IsList returns nil if the provided Object is a List kind.
+// If not, an error will be returned.
+func IsList(obj Object) error {
+	switch t := obj.(type) {
+	case *Unknown:
+		if !strings.HasSuffix(t.Kind, "List") {
+			return fmt.Errorf("object's kind field has no List suffix")
+		}
+		m := make(map[string]interface{})
+		if err := json.Unmarshal(t.RawJSON, &m); err != nil {
+			return err
+		}
+		items, exists := m["items"]
+		if !exists {
+			return fmt.Errorf("no items field in %#v", m)
+		}
+		_, ok := items.([]interface{})
+		if !ok {
+			return fmt.Errorf("object's items field isn't a slice")
+		}
+	case *Unstructured:
+		if !strings.HasSuffix(t.Kind, "List") {
```

Comment:
I don't want to force this based on name.  I'd rather do it based on our ability to find the data.